### ISSUE TITLE
Loading this webp image crashes

### ISSUE THREAD

**Issue by DevWhan** [SDWebImage/SDWebImageWebPCoder#114]
unused  SDWebImageAWebPCoder
[de3bc07f4186db0094ed2187883cd46e.zip](https://github.com/user-attachments/files/18229158/de3bc07f4186db0094ed2187883cd46e.zip)

**Comment by DevWhan** [comment]
In the latest SDWebImage Demo, it can be reproduced. I only used SDImageWebPCoder instead of SDImageAWebPCoder. I called sd_setImageWithURL and passed in the remote address of this image. The memory began to grow infinitely until it was killed.
iPhone11 iOS18.2

**Comment by DevWhan** [comment]
I tried to use SDWebImageAPWebCoder in our project and it also crashed. Using Demo, I finally found that the memory of this image would soar to 1.6G when loading, and if multiple images were displayed in the UITableViewCell, it would crash. I don't know if it was the problem with the image itself, or if the decoding was just so big. The Demo has been uploaded. If you have time to trouble reading it, thank you very much
[SDWebImage-5.20.0 2.zip](https://github.com/user-attachments/files/18235558/SDWebImage-5.20.0.2.zip)

**Comment by lperezdude** [comment]
@DevWhan  I think im running into the same issue. I have a feed paginated of images from a server. However when enough images are loaded they stop loading, app does not crash but i get the following on the warning on the console. 
Any idea how to stop this from happening ? 

"CGSImageDataLock: Cannot allocate memory
CGBitmapContextInfoCreate: unable to allocate 48771072 bytes for bitmap data
CGBitmapContextInfoCreate: unable to allocate 48771072 bytes for bitmap data
CGDisplayListDrawInContext: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
CGBitmapContextCreateImage: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
CGBitmapContextInfoCreate: unable to allocate 48771072 bytes for bitmap data
CGBitmapContextInfoCreate: unable to allocate 48771072 bytes for bitmap data
CGDisplayListDrawInContext: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable.
CGBitmapContextCreateImage: invalid context 0x0. If you want to see the backtrace, please set CG_CONTEXT_SHOW_BACKTRACE environmental variable."

**Comment by dreampiggy** [comment]
OOM ?

Maybe you can enable that `MaxBytesSize` option or using `ThumbnailPixelSize`

**Comment by DevWhan** [comment]
> OOM ?
> 
> Maybe you can use that `ScaleDownIimitBytesSize` (https://sdwebimage.github.io/documentation/sdwebimage/sdwebimagecontextoption/imagescaledownlimitbytes) option or using `ThumbnailPixelSize`
> 
> This can reduce the RAM usage for loading

Yes, I suspect that the OOM was caused by SDImageWebPCoder. The image itself is only 3.4MB
If you have time, you can look at the last demo I sent, and you can reproduce this problem.
ScaleDownIimitBytesSize can really solve my problem, thank you very much

**Comment by DevWhan** [comment]
thank
